velox/functions/sparksql/specialforms/SparkCastHooks.cpp

```diff
@@ -41,7 +44,7 @@ int32_t SparkCastHooks::castStringToDate(const StringView& dateString) const {
 }
 
 bool SparkCastHooks::legacy() const {
-  return false;
+  return legacyCast_;
```

Comment:
This hook is used by applyCastPrimitives, and the behavior of Spark does not vary with the isLegacyCast config. We may need to double-confirm this change.

Reply:
Yes, I found that the legacy_cast configuration semantics did not match the current implementation. To make this distinction, the `spark.legacy_cast_to_str` configuration has been added.  
> legacy_cast: Enables legacy CAST semantics if set to true. CAST(timestamp AS varchar) uses ‘T’ as separator between date and time (instead of a space), and the year part is not padded.

Reply:
@dcoliversun Is the new configuration added to adapt to different behaviors of Spark? I notice the implementation of Spark seems to be determined, could you clarify a bit why we need this config? Thanks.